protos/sources/vega/markets.proto

```diff
@@ -42,6 +42,17 @@ message Future {
 
   // Binding between the data spec and the data source.
   DataSourceSpecToFutureBinding data_source_spec_binding = 5;
+  // If set, this product represents a capped future market.
+  optional FutureCap cap = 6;
+}
+
+message FutureCap {
+  // Set the maximum price for orders, and settlement data in market decimals.
+  string max_price = 1;
+  // If set to true, the settlement price must either be zero, or equal to the max price.
+  optional bool binary_settlement = 2;
+  // If set to true, positions must be fully collateralised so there is no default risk for any party.
+  optional bool fully_collateralised = 3;
```

Comment:
do these need to be optional? they'd be false by default if not specificied both in JSON or proto? I mean, not sure a optional boolean makes sense by definition.

Reply:
It doesn't _quite_ make sense, but marked them as optional as they are described as optional in the spec, and in previous PR's there's been comments to mark any fields as optional if the spec calls for it.

Reply:
right.